addon/mixins/log-api-errors.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+    get,
+    canInvoke,
+    inject: { service }
+} = Ember;
+
+export default Ember.Mixin.create({
+    errorManager: service('error-manager'),
+
+    handleResponse(status, headers, payload, requestData) {
+        let result = this._super(...arguments);
+
+        if(this.isSuccess(status, headers, payload)) {
+            return result;
+        }
+
+        //provide a more detailed message (request/response info, ect) for ALL error types
+        result.message = this.formatErrorMessage(result, status, headers, payload, requestData);
+        get(this, 'errorManager').consume(result);
+
+        return result;
+    },
+
+    formatErrorMessage(error, status, headers, payload, requestData) {
+        if(typeof payload !== 'string') {
+            payload = JSON.stringify(payload);
+        }
+
+        if(canInvoke(this, 'generatedDetailedMessage')) {
```

Comment:
@billdami Shouldn't `canInvoke` be `tryInvoke` https://www.emberjs.com/api/#method_tryInvoke